Snakefile

```diff
@@ -0,0 +1,94 @@
+KSIZES = [21, 31, 51]                                                    # create a list of k-mer sizes for the workflow
+
+rule all:
+    input: expand("outputs/sourmash_sketch_subtract/{sample}_k{ksize}.sig", sample = SAMPLES, ksize = KSIZES)
+
+rule sourmash_sketch:
+    """
+    Use fastq-dump to download sequencing data for a given accession.
+    Pipe the sequencing data directly to a sketch without writing to disk.
+    Using fastq-dump like this dumps both single end and paired end reads into std out, which can then be piped into sourmash sketch
+    """
+    output: "outputs/sourmash_sketch/{run_accession}.sig"
+    conda: 'envs/sourmash.yml'
+    shell:'''
+    fastq-dump --disable-multithreading --fasta 0 --skip-technical --readids --read-filter pass --dumpbase --split-spot --clip -Z {wildcards.run_accession} | 
+        sourmash sketch dna -p k=21,k=31,k=51,scaled=200,abund --name {wildcards.run_accession} -o {output} -
+    '''
+
+rule calculate_mtx_not_in_mgx:
+    """
+    Use the sourmash python API to subtract metagenome fracminhash sketches from metatranscriptome fracminhash sketches.
+    This for loop re-implements a simplified veresion of the the CLI of sourmash sig subtract.
+    https://github.com/sourmash-bio/sourmash/blob/latest/src/sourmash/sig/__main__.py
+    Doing this in a python script is the easiest way to map pairs of samples to each other within the snakemake workflow.
+    
+    The run directive doesn't allow a run to have a conda environment; 
+    therefore the run environment for this code is specified in environment.yml
+    """
+    input:
+        metadata = 'inputs/metadata-paired-mgx-mtx.tsv',
+        sigs = expand('outputs/sourmash_sketch/{run_accession}.sig', run_accession = RUN_ACCESSIONS) 
+    output:
+        sigs = expand("outputs/sourmash_sketch_subtract/{sample}_k{{ksize}}.sig", sample = SAMPLES)
+    run:
+        ksize = wildcards.ksize
+        # read in metadata dataframe to derive sample pairs
+        metadata = pd.read_csv(input.metadata, sep = "\t") # read in metadata as pandas df
+        metadata = metadata.reset_index()  # make sure indexes pair with number of rows
+        
+        for index, row in metadata.iterrows():
+            # grab the run accessions for a given paired metagenome and metatranscriptome
+            mtx_run_accession = row['mtx_run_accession']
+            mgx_run_accession = row['mgx_run_accession']
+            # using the run assession, create paths of the signatures
+            mtx_sigfile = os.path.join('outputs/sourmash_sketch', mtx_run_accession + '.sig')
+            mgx_sigfile = os.path.join('outputs/sourmash_sketch', mgx_run_accession + '.sig')
+
+            # read in mtx signature and grab both the minhash object and the hashes in that object
+            mtx_sigobj = sourmash.load_one_signature(mtx_sigfile, ksize=ksize, select_moltype='DNA')
```

Comment:
I am assuming `select_moltype = 'DNA'` actually means nucleic acid here and is appropriate for mtx?

Reply:
yes!! the options are DNA or protein, where DNA actually specifies nucleotide